pkg/controllers/status/common.go

```diff
@@ -105,43 +108,63 @@ func updateResourceStatus(
 	dynamicClient dynamic.Interface,
 	restMapper meta.RESTMapper,
 	interpreter resourceinterpreter.ResourceInterpreter,
-	resource *unstructured.Unstructured,
+	eventRecorder record.EventRecorder,
+	objRef workv1alpha2.ObjectReference,
 	bindingStatus workv1alpha2.ResourceBindingStatus,
 ) error {
-	gvr, err := restmapper.GetGroupVersionResource(restMapper, schema.FromAPIVersionAndKind(resource.GetAPIVersion(), resource.GetKind()))
+	gvr, err := restmapper.GetGroupVersionResource(restMapper, schema.FromAPIVersionAndKind(objRef.APIVersion, objRef.Kind))
 	if err != nil {
-		klog.Errorf("Failed to get GVR from GVK(%s/%s), Error: %v", resource.GetAPIVersion(), resource.GetKind(), err)
+		klog.Errorf("Failed to get GVR from GVK(%s/%s), Error: %v", objRef.APIVersion, objRef.Kind, err)
 		return err
 	}
 
-	if !interpreter.HookEnabled(resource.GroupVersionKind(), configv1alpha1.InterpreterOperationAggregateStatus) {
+	gvk := schema.GroupVersionKind{Group: gvr.Group, Version: gvr.Version, Kind: objRef.Kind}
+	if !interpreter.HookEnabled(gvk, configv1alpha1.InterpreterOperationAggregateStatus) {
 		return nil
 	}
-	newObj, err := interpreter.AggregateStatus(resource, bindingStatus.AggregatedStatus)
-	if err != nil {
-		klog.Errorf("Failed to aggregate status for resource(%s/%s/%s, Error: %v", gvr, resource.GetNamespace(), resource.GetName(), err)
-		return err
-	}
 
-	oldStatus, _, _ := unstructured.NestedFieldNoCopy(resource.Object, "status")
-	newStatus, _, _ := unstructured.NestedFieldNoCopy(newObj.Object, "status")
-	if reflect.DeepEqual(oldStatus, newStatus) {
-		klog.V(3).Infof("Ignore update resource(%s/%s/%s) status as up to date.", gvr, resource.GetNamespace(), resource.GetName())
-		return nil
-	}
+	resource := &unstructured.Unstructured{}
+	if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// get resource template directly from apiserver, rather than informer cache, detail refers to:
+		// https://github.com/karmada-io/karmada/issues/5285
```

Comment:
```suggestion
		// Fetch resource template from karmada-apiserver instead of informer cache, to avoid retry due to
		// resource conflict which often happens, especially with a huge amount of resource templates and
		// the informer cache doesn't sync quickly enough.
		// For more details refer to https://github.com/karmada-io/karmada/issues/5285.
```